src/plugins/visualize/public/plugin.ts

```diff
@@ -114,6 +115,15 @@ export class VisualizePlugin
         },
       ],
       getHistory: () => this.currentHistory!,
+      onBeforeNavLinkSaved: (urlToSave: string) => {
+        if (
+          !urlToSave.includes(`${VisualizeConstants.EDIT_PATH}/`) &&
+          this.isLinkedToOriginatingApp?.()
+        ) {
+          return core.http.basePath.prepend('/app/visualize');
```

Comment:
I'm surprised this isn't pulled out into a separate constant of some kind, in case it ever changes or becomes configurable

Reply:
I was surprised about this too - though it was only used once before this PR so maybe this is the right time to pull it out. Will add it to the constants.